daemon/exec.go

```diff
@@ -248,3 +248,34 @@ func (d *Daemon) Exec(c *Container, execConfig *execConfig, pipes *execdriver.Pi
 
 	return exitStatus, err
 }
+
+// execCommandGC starts a go routine on a ticker to clean up the daemon references
+// of exec configs that are no longer part of the container.
+func (d *Daemon) execCommandGC() {
+	for range time.Tick(5 * time.Minute) {
+		var (
+			cleaned          int
+			liveExecCommands = d.containerExecIds()
+			ids              = d.execCommands.List()
+		)
+		for _, id := range ids {
+			if _, exists := liveExecCommands[id]; !exists {
```

Comment:
@duglin no, the locks will be held on the read/writes and it wont happen.  Since the baseline for the GC is the container's list and immediate deletes happen on that list it is synchronized.